### DISCUSSION TITLE
Components & Dynamic zones

### DISCUSSION THREAD

**Discussion by iksent** [directus/directus#3707] [Archive]
Hello, I am [Strapi](https://github.com/strapi/strapi) user.

Looking forward to your 9th release, just tried it and it looks amazing - the design is great and was made very accurate!

But can you please tell me, does Directus have support of components (nested fields can be organized in groups and then reused in any model) and Dynamic zones?

This is a perfect features from Strapi: https://strapi.io/blog/release-beta-18-dynamic-zones

The most popular case for me is to create component "SEO" with 3 fields like "title", "description" and "image" and then use it for all single models = pages.

**Comment by rijkvanzanten** [comment]
> Hello, I am Strapi user.

Hello! Welcome

> But can you please tell me, does Directus have support of components (nested fields can be organized in groups and then reused in any model) and Dynamic zones?

Yup! Nested fields/components can be achieved through what we call relationships (many-to-one, one-to-many, many-to-many). Directus' equivalent of "Dynamic Zones" would be the many-to-any relationship w/ the Builder interface.

We have a server on Discord if you have more questions 👍🏻 https://directus.chat

**Reply by cgx9** [reply]
> https://directus.chat



> Yup! Nested fields/components can be achieved through what we call relationships (many-to-one, one-to-many, many-to-many). Directus' equivalent of "Dynamic Zones" would be the many-to-any relationship w/ the Builder interface.

I use json type data and then select Repeater in the interface and then add new fields to achieve similar functionality. But only standard fields can be selected, I need to have fields of image type. Because I design an item sku, it contains color, size, image, price etc.

**Reply by rijkvanzanten** [reply]
@cgx9 Yeah, you can't use static JSON if you need dynamically nested data 👍🏻

**Comment by iksent** [comment]
Thanks a lot for the fast answer!

Works good, except that I can't understand: is it possible to define the order of many-to-many elements **for the API/GraphQL**: 

![image](https://user-images.githubusercontent.com/32956521/104758481-547f6e80-5780-11eb-8994-407b6cb141a5.png)

I am trying to order them within the left icon but order resets after page refresh or data refetch.

**Reply by benhaynes** [reply]
Not sure why it would reset after page refresh unless you didn't save the parent or are manually sorting by another column in that interface. To be clear, are you talking about the order of items in the App or API?

**Reply by iksent** [reply]
I am talking about API, so I can change the order of elements in Admin and then get these elements in the same order via the API or GraphQL

**Reply by benhaynes** [reply]
Ahh, I believe you would want to use the `sort` parameter then. I forget the exact syntax for sorting relational data (not the parent)... @rijkvanzanten ? Let's get that added to the Docs today too... 👍
